services/reddit/subreddit-subscribers.service.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const { BaseJsonService } = require('..')
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { metric } = require('../text-formatters')
+
+const schema = Joi.object({ subscribers: nonNegativeInteger })
```

Comment:
I believe that the `subscribers` field is contained within the `data` object key in the response isn't it?

For example
```json
{
  "kind":"t5",
  "data": {
      ...
     "subscribers": x,
      ...
   }
}
```

Reply:
Yes, I'm filtering the response here:
`async handle({ subreddit }) {`
`const json = await this.fetch({ subreddit })`
`return this.constructor.render({`
`subreddit,`
`subscribers: json.data.subscribers,`
`})`
}`

Am I doing it wrong?

Reply:
The schema is used to validate the raw response, but you've currently defined the schema structured to expect the `subscribers` field on the root response object. That does not match the actual response, so it's being rejected and the badge is not working:

https://shields-staging-pr-3443.herokuapp.com/reddit/subreddit-subscribers/drums.svg
![](https://shields-staging-pr-3443.herokuapp.com/reddit/subreddit-subscribers/drums.svg)

You'll need to update the schema to match the response, which I think would be something like this:
```js
const schema = Joi.object({
  data: Joi.object({
    subscribers: nonNegativeInteger,
  }).required()
}).required()
```

Reply:
You're right, now it's working properly. I thought the `schema` worked in another way. Pushing the new version. Thanks!

Reply:
👍